core/src/main/scala/caliban/validation/Validator.scala

```diff
@@ -132,7 +133,7 @@ object Validator {
   private def collectSelectionSets(selectionSet: List[Selection]): List[Selection] =
     selectionSet ++ selectionSet.flatMap {
       case _: FragmentSpread                  => Nil
-      case Field(_, _, _, _, selectionSet)    => collectSelectionSets(selectionSet)
+      case Field(_, _, _, _, selectionSet, _) => collectSelectionSets(selectionSet)
```

Comment:
I'd be inclined to use a more concise notation that does not use underscores much, once again here.
`case f: Field => collectSelectionSets(f.selectionSet)`

Reply:
Here should be fine since exhaustive checking would already be disabled from the line above